test/zdtm/static/cgroup_ignore.c

```diff
@@ -0,0 +1,157 @@
+#include <unistd.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/mount.h>
+#include "zdtmtst.h"
+
+const char *test_doc = "Check that cgroups are correctly ignored";
+const char *test_author = "Adrian Reber <areber@redhat.com>";
+
+char *dirname;
+TEST_OPTION(dirname, string, "cgroup directory name", 1);
+static const char *cgname = "zdtmtst";
+
+static size_t read_all(int fd, char *buf, size_t size)
+{
+	ssize_t r = 0, ret;
+
+	while (r < size) {
+		ret = read(fd, buf + r, size - r);
+		if (ret < 0) {
+			pr_perror("Read failed");
+			return -1;
+		} else if (ret == 0) {
+			return 0;
+		}
+		r += ret;
+	}
+
+	return 0;
+}
+
+int main(int argc, char **argv)
+{
+	cleanup_close int fd = -1;
+	cleanup_close int cgroup_procs_fd = -1;
+	cleanup_free char *buffer_old = NULL;
+	cleanup_free char *buffer_new = NULL;
+	cleanup_free char *destination = NULL;
+	cleanup_free char *cgroup_procs = NULL;
+	cleanup_free char *own_pid = NULL;
```

Comment:
imho: cleanup_free doesn't look good. In this case, you probably can use alloca(), or declare fixed size arrays:
char buffer_new[PAGE_SIZE];